lib/src/main/java/io/ably/lib/transport/WebSocketTransport.java

```diff
@@ -147,8 +150,29 @@ class WsClient extends WebSocketClient {
         @Override
         public void onOpen(ServerHandshake handshakedata) {
             Log.d(TAG, "onOpen()");
-            connectListener.onTransportAvailable(WebSocketTransport.this);
-            flagActivity();
+            if (isHostnameVerified(params.host)) {
+                connectListener.onTransportAvailable(WebSocketTransport.this);
+                flagActivity();
+            } else {
+                connectListener.onTransportUnavailable(WebSocketTransport.this, ConnectionManager.REASON_REFUSED);
+                close();
+            }
+        }
+
+        /**
+         * Added because we had to override the onSetSSLParameters() that usually performs this verification.
+         * When the minSdkVersion will be updated to 24 we should remove this method and its usages.
+         * https://github.com/TooTallNate/Java-WebSocket/wiki/No-such-method-error-setEndpointIdentificationAlgorithm#workaround
+         */
+        private boolean isHostnameVerified(String hostname) {
+            SSLSession session = getSSLSession();
+            if (HttpsURLConnection.getDefaultHostnameVerifier().verify(hostname, session)) {
+                Log.i(TAG, "Successfully verified hostname");
```

Comment:
I think `i` is too high - I would use `d` or even `v` here.

Reply:
https://github.com/ably/ably-java/pull/808/commits/8831ae3d391b000e48ae961701a8069a3fb6816f